src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -450,112 +449,116 @@ protected virtual string GenerateUniqueId()
+                var currentWriteAttributeMethodName = _context.Host.GeneratedClassContext.WriteAttributeValueMethodName;
+                _context.TargetWriterName = null;
 
-                    // TagHelper attribute rendering is buffered by default. We do not want to write to the current
-                    // writer.
-                    var currentTargetWriter = _context.TargetWriterName;
-                    var currentWriteAttributeMethodName = _context.Host.GeneratedClassContext.WriteAttributeValueMethodName;
-                    _context.TargetWriterName = null;
+                Debug.Assert(attribute.Value is ParentChunk);
+                var children = ((ParentChunk)attribute.Value).Children;
+                var attributeCount = children.Count(c => c is DynamicCodeAttributeChunk || c is LiteralCodeAttributeChunk);
 
-                    Debug.Assert(attributeValueChunk is ParentChunk);
-                    var children = ((ParentChunk)attributeValueChunk).Children;
-                    var attributeCount = children.Count(c => c is DynamicCodeAttributeChunk || c is LiteralCodeAttributeChunk);
+                _writer
+                    .WriteStartMethodInvocation(_tagHelperContext.BeginAddHtmlAttributeValuesMethodName)
+                    .Write(ExecutionContextVariableName)
+                    .WriteParameterSeparator()
+                    .WriteStringLiteral(attribute.Name)
+                    .WriteParameterSeparator()
+                    .Write(attributeCount.ToString(CultureInfo.InvariantCulture))
+                    .WriteParameterSeparator()
+                    .Write(attributeStructureParameter)
+                    .WriteEndMethodInvocation();
 
-                    _writer
-                        .WriteStartMethodInvocation(_tagHelperContext.BeginAddHtmlAttributeValuesMethodName)
-                        .Write(ExecutionContextVariableName)
-                        .WriteParameterSeparator()
-                        .WriteStringLiteral(attributeName)
-                        .WriteParameterSeparator()
-                        .Write(attributeCount.ToString(CultureInfo.InvariantCulture))
-                        .WriteEndMethodInvocation();
+                _attributeCodeVisitor.Accept(attribute.Value);
 
-                    _attributeCodeVisitor.Accept(attributeValueChunk);
+                _writer.WriteMethodInvocation(
+                    _tagHelperContext.EndAddHtmlAttributeValuesMethodName,
+                    ExecutionContextVariableName);
 
-                    _writer.WriteMethodInvocation(
-                        _tagHelperContext.EndAddHtmlAttributeValuesMethodName,
-                        ExecutionContextVariableName);
+                _context.TargetWriterName = currentTargetWriter;
+            }
+            else if (attribute.Structure == HtmlAttributeStructure.Minimized ||
+                TryGetPlainTextValue(attribute.Value, out textValue))
```

Comment:
Not a fan of generating more complex code for the simple `attribute.Structure == HtmlAttributeStructure.Minimized` case.